Core/Core/Transports/ServerV2.cs

```diff
@@ -415,7 +398,13 @@ private async void SendingThreadMain()
           _sendBuffer.Clear();
           _exception = ex;
         }
-        return;
+
```

Comment:
Okay but this whole class has some issue as you are probably aware. locking will hold the thread and not await, thread.sleep similarly, should be Task.Delay() which is awaitable. I presume there is some queue here and it may be we can engineer this a bit better.

Reply:
Well aware of the async related issues with our transports.
I'm not a fan of the implementation, but gonna require some more involved refactors to address.

(see https://spockle.atlassian.net/browse/CNX-8613)

For now, resolving this item, it's not something we can solve right away.